apache/superset
# feat: validation db modal

### SUMMARY
This PR uses the validation api and the new LabeledErrorBoundInput component to validate a person's db connection as they are filling out the form. The validation happens on blur. 

We plan to add some client side validation on types such as checking that hostnames don't have spaces and port is a number, etc. That will come in a separate PR. For now, all of this work is not visible to the user. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/5186919/119589260-4c1aee00-bd87-11eb-8914-129f58e9ed8d.mov

I left the network requests tab open so that we can evaluate how many requests are made during this flow, to see if it needs to be optimized. We could evaluate if anything changed on blur if needed. 

### TESTING INSTRUCTIONS
For now, let's make sure that this code doesn't break anything existing. Try creating a database and editing one. 

Because of how RTL tests visually and we can't access this feature unless by hard-coding a param, we'll add more tests in a future PR before making this visible.


### ADDITIONAL INFORMATION
